Lykke.B2c2Client/B2С2RestClient.cs

```diff
@@ -277,6 +282,194 @@ public async Task<List<LedgerLog>> GetLedgerHistoryAsync(int offset = 0, int lim
             }
         }
 
+        public async Task<PaginationResponse<List<LedgerLog>>> GetLedgerHistoryAsync(LedgersRequest request, CancellationToken ct = default(CancellationToken))
+        {
+            var requestId = Guid.NewGuid();
+            _log.Info("ledger history - request", context: $"requestId: {requestId}, request: {request?.ToJson()}");
+
+            try
+            {
+                var param = new Dictionary<string, string>();
+
+                if (request != null)
+                {
+                    if (request.CreatedAfter.HasValue)
+                    {
+                        param.Add("created__gte", request.CreatedAfter.Value.ToString("yyyy-MM-ddThh:mm:ss"));
+                    }
+
+                    if (request.CreatedBefore.HasValue)
+                    {
+                        param.Add("created__lt", request.CreatedBefore.Value.ToString("yyyy-MM-ddThh:mm:ss"));
+                    }
+
+                    if (!string.IsNullOrEmpty(request.Currency))
+                    {
+                        param.Add("currency", request.Currency);
+                    }
+
+                    if (request.Type.HasValue)
+                    {
+                        param.Add("type", request.Type.ToString());
+                    }
+
+                    if (request.Since.HasValue)
+                    {
+                        param.Add("since", request.Since.Value.ToString("yyyy-MM-ddThh:mm:ss"));
+                    }
+
+                    if (request.Offset > 0)
+                    {
+                        param.Add("offset", Math.Max(50000, request.Offset).ToString());
```

Comment:
In the case, if the factual offset is more the 50000 the cycle will stuck, so better to write error to the log and break the cycle.